tests/test_router.py

```diff
@@ -174,3 +152,43 @@ def sample():
     expected_query = {"hello": ["world"], "thing": ["1", "2"]}
     await display.goto("?hello=world&thing=1&thing=2")
     await display.page.wait_for_selector("#success")
+
+
+async def test_custom_path_compiler(display: DisplayFixture):
+    expected_params = {}
+
+    @component
+    def check_params():
+        assert use_params() == expected_params
+        return html.h1({"id": "success"}, "success")
+
+    @component
+    def sample():
+        return router(
+            Route(
+                r"/first/(?P<first__int>\d+)",
+                check_params(),
+                Route(
+                    r"/second/(?P<second__float>[\d\.]+)",
+                    check_params(),
+                    Route(
+                        r"/third/(?P<third__list>[\d,]+)",
+                        check_params(),
+                    ),
+                ),
+            ),
+            compiler=compile_simple_regex_route,
```

Comment:
This should not be configured globally. While unlikely, it's conceivable that one could use different routing packages in the same project. We can supply a `router = create_router(my_compiler)` instead.

Reply:
Not a global override, but a global default.

We don't want Django users to have to set the compiler parameter every time.

Reply:
If library A expects the default compiler to be X and library B configures the default compiler to be Y then library A breaks. I don't follow how this would work without some sort of compiler context.

I think the Django-specific router is just something that would be imported from `reactpy_django` - no configuration required by users.

Reply:
For context, React Router has multiple router implementations and uses [create router functions](https://reactrouter.com/en/main/routers/create-browser-router) for each type of router they provide.

Reply:
Looking at the [parameters](https://github.com/remix-run/react-router/blob/2cd19443cb772f2b10c4858a40dc9343546c5b5f/packages/router/router.ts#L344) accepted by that function reminds me that a `use_history` hook might be useful.

Reply:
Yep, `useHistory` was one of the [original things](https://github.com/idom-team/idom-router/issues/2) I recommended porting over

Reply:
Can we create an issue for hooks that we would like to implement?